### ISSUE TITLE
Flutter 3 changes

### ISSUE THREAD

**Issue by Sesa1988** [LinusU/flutter_web_auth#117]
Flutters API for widget binding changed and we need probably a small update to prevent a failed build.
Should probably be the same error for all platforms.

<img width="1294" alt="image" src="https://user-images.githubusercontent.com/26410357/167947697-f91da752-bb81-4063-abb7-2c67bb42d1b4.png">

**Comment by rienkkk** [comment]
Same error. The ! should be removed from WidgetBinding.instance. Since Flutter 3.0 it can't be null anymore.

**Comment by noga-dev** [comment]
You can override the pubspec to use my branch for now while we wait for #118 to be merged:

```yaml
flutter_web_auth:
  git:
     url: https://github.com/noga-dev/flutter_web_auth.git
     ref: ee3c26a530a625c3cab7e0c45cc30316c8b4b64c
```

**Comment by AndrewStickler** [comment]
Any chance of an official fix for this soon please?

**Comment by AndrewStickler** [comment]
Actually, I see this package has now been superseded by flutter_web_auth2 which doesn't have that problem, so please ignore my request.

**Comment by LinusU** [comment]
This is now fixed in [🚢 0.4.2 / 2022-11-01](https://github.com/LinusU/flutter_web_auth/releases/tag/0.4.2)

**Comment by noga-dev** [comment]
> This is now fixed in [🚢 0.4.2 / 2022-11-01](https://github.com/LinusU/flutter_web_auth/releases/tag/0.4.2)

Hi Linus! It's good to know you're alive.

With that in mind:

The PR you've accepted jut now (#132) is a dupe opened over a month after #118 and lacks some extras from it.

And as I've already migrated to @ThexXTURBOXx 's fork, I would prefer to stick with it (unless he'll stop maintaining it) as he's been doing a fantastic job that deserves acknowledgement. Would you be open to letting him take over (or making him a contributor)?

Cheers and welcome back.

**Comment by LinusU** [comment]
Thanks!

Yes, that was intentional since I didn't want to force Flutter 3.0 in a minor version. Other than the changed flutter version range.

I think that you can absolutely stick with the fork if that works for you 👍

**Comment by noga-dev** [comment]
> This had do be reverted since it broke Flutter v2 compatibility, it's now released as [🚢 0.5.0 / 2022-11-02](https://github.com/LinusU/flutter_web_auth/releases/tag/0.5.0) instead!

Among the extras in #118 was the version constraint.

**Comment by LinusU** [comment]
@noga-dev, yes. I didn't realise that removing the `?` wasn't a backwards compatible change. Instead, I though that the version constraint wasn't needed, and that adding it would have made it a breaking change.

Sorry for not merging your PR, had I understood that it was indeed a breaking change I would absolutely have done that...

**Comment by noga-dev** [comment]
> @noga-dev, yes. I didn't realise that removing the `?` wasn't a backwards compatible change. Instead, I though that the version constraint wasn't needed, and that adding it would have made it a breaking change.
> 
> Sorry for not merging your PR, had I understood that it was indeed a breaking change I would absolutely have done that...

Definitely an easy thing to miss. Thank you for the amazing work with the package. 🙏
